referenceframe/frame.go

```diff
@@ -49,11 +52,13 @@ func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64, nodePos
 			u = 999
 		}
 
-		span := u - l
-		l = nodePose[i].Value - span/2
-		pos = append(pos, Input{lim*span*rSeed.Float64() + l + (span * (1 - lim) / 2)})
+		frameSpan := u - l
+		minVal := math.Max(l, nodeLocation[i].Value - frameSpan/2)
```

Comment:
I believe both of these should be `lim * frameSpan/2`, such that `samplingSpan` is of size `lim * frameSpan`.

Then below we can do `pos = append(pos, Input{samplingSpan*rSeed.Float64() + minVal})

Reply:
ok makes sense, thanks